InterviewQuestions/Chapters/CustomLinkedLists/CustomLinkedList.cs

```diff
@@ -1,39 +1,69 @@
-﻿namespace InterviewQuestions.Chapters.CustomLinkedLists
-{
-    public class CustomLinkedList
-    {
-        public class Node
-        {
-            public int data;
-            public Node? next;
-            public Node(int data)
-            {
-                this.data = data;
-                this.next = null;
-            }
-        }
-        public Node? head;
-
-        public CustomLinkedList()
-        {
-            head = null;
-        }
-        public void add(int newElement)
-        {
-            Node newNode = new Node(newElement);
-            if (head == null)
-            {
-                head = newNode;
-            }
-            else
-            {
-                Node currentNode = head;
-                while (currentNode.next != null)
-                {
-                    currentNode = currentNode.next;
-                }
-                currentNode.next = newNode;
-            }
-        }
-    }
-}
+﻿using System.Diagnostics;
+using System.Security;
+
+namespace InterviewQuestions.Chapters.CustomLinkedLists
+{
+    public class CustomLinkedList
+    {
+        public class Node
+        {
+            public int data;
+            public Node? next;
+            public Node(int data)
+            {
+                this.data = data;
+                this.next = null;
+            }
+        }
+        public Node? head;
+
+        public CustomLinkedList()
+        {
+            head = null;
+        }
+        public void add(int newElement)
+        {
+            Node newNode = new Node(newElement);
+            if (head == null)
+            {
+                head = newNode;
+            }
+            else
+            {
+                Node currentNode = head;
+                while (currentNode.next != null)
+                {
+                    currentNode = currentNode.next;
+                }
+                currentNode.next = newNode;
+            }
+        }
+        public int getKhtLastElement(int index)
+        {
+            Node? currentNode = head;
+            int lenght = 0;
+            while (currentNode != null) 
+            {
+                currentNode = currentNode.next;
+                lenght++;
+            }
+            if (lenght < index)
+            {
+                Debug.Assert(false);

```

Comment:
instead of debug.assert, throw an exception here. and also write unit tests for it. It would be good practice to check if your method throws an exception on the bad cases.

Reply:
throw an exception instead of debug.assert. look here 9255e5871f9a8e4e3cd09e462367b94ad4ba77f4